### ISSUE TITLE
Problem with LeakCanary

### ISSUE THREAD

**Issue by Piasy** [Piasy/OkBuck#20]
After build and install the apk, the app crashed with following message:
```java
FATAL EXCEPTION: pool-3-thread-1
Process: com.github.piasy.template, PID: 23363
java.lang.IllegalArgumentException: Component class com.squareup.leakcanary.internal.DisplayLeakActivity does not exist in com.github.piasy.template
    at android.os.Parcel.readException(Parcel.java:1476)
    at android.os.Parcel.readException(Parcel.java:1426)
    at android.content.pm.IPackageManager$Stub$Proxy.setComponentEnabledSetting(IPackageManager.java:2943)
    at android.app.ApplicationPackageManager.setComponentEnabledSetting(ApplicationPackageManager.java:1285)
    at com.squareup.leakcanary.internal.LeakCanaryInternals$1.run(LeakCanaryInternals.java:105)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
    at java.lang.Thread.run(Thread.java:841)

```

**Comment by Piasy** [comment]
This is because the LeakCanary library have declare some component in its AndroidManifest.xml file, which will be merged into the final AndroidManifest.xml in gradle build, while BUCK build will not merge AndroidManifest.xml file.

I think this problem could be fixed by doing this merge work during BUCK, through adding some build rules into BUCK script.

**Comment by Piasy** [comment]
And the merged manifest is:
```xml
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="com.squareup.leakcanary" >

    <uses-sdk
        android:minSdkVersion="8"
        android:targetSdkVersion="21" />

    <!-- To store the heap dumps and leak analysis results. -->
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />

    <application>
        <service
            android:name="com.squareup.leakcanary.internal.HeapAnalyzerService"
            android:enabled="false"
            android:process=":leakcanary" />
        <service
            android:name="com.squareup.leakcanary.DisplayLeakService"
            android:enabled="false" />

        <activity
            android:name="com.squareup.leakcanary.internal.DisplayLeakActivity"
            android:enabled="false"
            android:icon="@drawable/__leak_canary_icon"
            android:label="@string/__leak_canary_display_activity_label"
            android:taskAffinity="com.squareup.leakcanary"
            android:theme="@style/__LeakCanary.Base" >
            <intent-filter>
                <action android:name="android.intent.action.MAIN" />

                <category android:name="android.intent.category.LAUNCHER" />
            </intent-filter>
        </activity>
    </application>

</manifest>
```

With LeakCanary 1.3.1.

**Comment by Piasy** [comment]
It will be fixed at #25

**Comment by Piasy** [comment]
fixed at #36
